From 48c7a1aa186ae46e53710c5b57945d46e727ad14 Mon Sep 17 00:00:00 2001
From: Julian Bouzas <julian.bouzas@collabora.com>
Date: Thu, 4 Jun 2026 08:43:04 -0400
Subject: [PATCH] state-stream: Only use 'media.role' key if its value is
 'Notification' when forming key

This avoids different applications sharing the same role to have the same volume
at startup, unless those are part of the Notification role.

Fixes #955

diff --git a/src/scripts/node/state-stream.lua b/src/scripts/node/state-stream.lua
index 6587f26b..652d97a2 100644
--- a/src/scripts/node/state-stream.lua
+++ b/src/scripts/node/state-stream.lua
@@ -399,7 +399,6 @@ end
 
 function formKey (properties)
   local keys = {
-    "media.role",
     "application.id",
     "application.name",
     "media.name",
@@ -407,14 +406,22 @@ function formKey (properties)
   }
   local key_base = nil
 
-  for _, k in ipairs (keys) do
-    local p = properties [k]
-    if p then
-      key_base = string.format ("%s:%s:%s",
-          properties ["media.class"]:gsub ("^Stream/", ""), k, p)
-      break
+  -- Only use the media.role key if it is set to Notification, otherwise use
+  -- the application keys.
+  if properties["media.role"] == "Notification" then
+    key_base = string.format ("%s:%s",
+        properties ["media.class"]:gsub ("^Stream/", ""), "media.role:Notification")
+  else
+    for _, k in ipairs (keys) do
+      local p = properties [k]
+      if p then
+        key_base = string.format ("%s:%s:%s",
+            properties ["media.class"]:gsub ("^Stream/", ""), k, p)
+        break
+      end
     end
   end
+
   return key_base
 end

